File: backend/src/main/java/com/smiles/facilities/domain/Facility.java
package com.smiles.facilities.domain;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.util.UUID;

/**
 * Facility entity representing a dental clinic/facility.
 */
@Entity
@Table(name = "facility")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Facility {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, unique = true)
    private String name;

    @Column(nullable = false, length = 100)
    private String city;

    @Column(nullable = false, length = 500)
    private String address;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;
}

File: frontend/src/types/facility.ts
/**
 * Facility entity representing a dental clinic/facility.
 */
export interface Facility {
  id: string;
  name: string;
  city: string;
  address: string;
  createdAt: string;
  updatedAt: string;
}

/**
 * Request DTO for creating a facility.
 */
export interface CreateFacilityRequest {
  name: string;
  city: string;
  address: string;
}

/**
 * Request DTO for updating a facility.
 */
export interface UpdateFacilityRequest {
  name?: string;
  city?: string;
  address?: string;
}
